File: frontend/src/lib/testUtils/fakers/ongoingGameFaker.ts
import { OngoingGame } from "@/features/lobby/lib/types";
import { faker } from "@faker-js/faker";
import { createFakePoolKey } from "./poolKeyFaker";
import { createFakeMinimalProfile } from "./minimalProfileFaker";

export function createFakeOngoingGame(
    overrides?: Partial<OngoingGame>,
): OngoingGame {
    return {
        gameToken: faker.string.alphanumeric(16),
        pool: createFakePoolKey(),
        opponent: createFakeMinimalProfile(),
        ...overrides,
    };
}

File: backend/Tests/AnarchyChess.Api.TestInfrastructure/Fakes/OngoingGameFaker.cs
using AnarchyChess.Api.Game.Models;
using AnarchyChess.Api.Matchmaking.Models;

namespace AnarchyChess.Api.TestInfrastructure.Fakes;

public class OngoingGameFaker : RecordFaker<OngoingGame>
{
    public OngoingGameFaker(PoolKey? poolKey = null)
    {
        StrictMode(true);
        RuleFor(x => x.GameToken, f => (GameToken)f.Random.AlphaNumeric(16));
        RuleFor(x => x.Pool, f => poolKey ?? new PoolKeyFaker().Generate());
        RuleFor(x => x.Opponent, f => new MinimalProfileFaker().Generate());
    }
}
